cmake/3rdparty.cmake

```diff
@@ -39,6 +39,10 @@ endif()
 
 set(CUB_DIR ${PROJECT_SOURCE_DIR}/3rdparty/cub CACHE STRING
     "Path to cub repo")
+add_library(cub INTERFACE IMPORTED)
+#target_include_directories(cub INTERFACE ${CUB_DIR}>) does not work with
+#cmake before 3.11, use the following for now:
```

Comment:
Do we have a specific version for requiring 3.10.2? I assume it was the default on some widely used distribution...

Reply:
@tijyojwad @vellamike ?

Reply:
Yep, it's default on Ubuntu 18.04 and provided all features we needed at that time, so we decided to stick to that. I'm okay with updating the version if deemed necessary

Reply:
Oh, ok, in that sense I think it makes sense to keep it for a little longer. I thought that came from Ubuntu 16.04